modules/yugabytedb/src/main/java/org/testcontainers/containers/YugabyteDBYCQLContainer.java

```diff
@@ -0,0 +1,168 @@
+package org.testcontainers.containers;
+
+import java.net.InetSocketAddress;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.Set;
+
+import com.github.dockerjava.api.command.InspectContainerResponse;
+import org.testcontainers.containers.delegate.YugabyteDBYCQLDelegate;
+import org.testcontainers.containers.strategy.YugabyteDBYCQLWaitStrategy;
+import org.testcontainers.ext.ScriptUtils;
+import org.testcontainers.utility.DockerImageName;
+
+/**
+ * Testcontainers implementation for YugabyteDB YCQL API.
+ *
+ * @author srinivasa-vasu
+ * @see <a href="https://docs.yugabyte.com/stable/api/ycql/">YCQL API</a>
+ */
+public class YugabyteDBYCQLContainer extends GenericContainer<YugabyteDBYCQLContainer> {
+
+	private static final DockerImageName DEFAULT_IMAGE_NAME = DockerImageName.parse("yugabytedb/yugabyte");
+
+	private static final Integer YCQL_PORT = 9042;
+
+	private static final Integer MASTER_DASHBOARD_PORT = 7000;
+
+	private static final Integer TSERVER_DASHBOARD_PORT = 9000;
+
+	private static final String ENTRYPOINT = "bin/yugabyted start --daemon=false";
+
+	private static final String LOCAL_DC = "datacenter1";
+
+	private String keyspace;
+
+	private String username;
+
+	private String password;
+
+	private String initScript;
+
+	/**
+	 * @param imageName image name
+	 */
+	public YugabyteDBYCQLContainer(final String imageName) {
+		this(DockerImageName.parse(imageName));
+	}
+
+	/**
+	 * @param imageName image name
+	 */
+	public YugabyteDBYCQLContainer(final DockerImageName imageName) {
+		super(imageName);
+		imageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
+		withExposedPorts(YCQL_PORT, MASTER_DASHBOARD_PORT, TSERVER_DASHBOARD_PORT);
+		waitingFor(new YugabyteDBYCQLWaitStrategy(this).withStartupTimeout(Duration.ofSeconds(60)));
+		withCommand(ENTRYPOINT);
+	}
+
+	@Override
+	public Set<Integer> getLivenessCheckPortNumbers() {
+		return Collections.singleton(getMappedPort(YCQL_PORT));
+	}
+
+	/**
+	 * Configures the environment variables. Setting up these variables would create the
+	 * custom objects. Setting {@link #withKeyspaceName(String)},
+	 * {@link #withUsername(String)}, {@link #withPassword(String)} these parameters will
+	 * initilaize the database with those custom values
+	 */
+	@Override
+	protected void configure() {
+		addEnv("YCQL_KEYSPACE", keyspace);
+		addEnv("YCQL_USER", username);
+		addEnv("YCQL_PASSWORD", password);
+	}
+
+	/**
+	 * @param initScript path of the initialization script file
+	 * @return {@link YugabyteDBYCQLContainer} instance
+	 */
+	public YugabyteDBYCQLContainer withInitScript(String initScript) {
+		this.initScript = initScript;
+		return this;
+	}
```

Comment:
I think this should not be supported because Testcontainers responsibility is about taking care of the service initialization. There are other tools which can help about keyspace and data management such as [liquibase-cassandra](https://docs.liquibase.com/install/tutorials/cassandra.html). Also, that way we are avoiding being tight to a Cassandra library version. @kiview WDYT?

Reply:
I agree with @eddumelendez. Adding such an API increases complexity more than necessary.

Reply:
I've decoupled the Cassandra client library dependency but kept the API as is. This API may be useful for standalone services. Let me know WDYT?